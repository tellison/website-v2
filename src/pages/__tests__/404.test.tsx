import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'vitest-axe';
import { describe, expect, it } from 'vitest'
import NotFound, { Head } from '../404';

describe('404 page', () => {
  it('renders correctly', () => {
    const { container } = render(<NotFound />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    const { container } = render(<Head />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('404: Not found | Adoptium');
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<NotFound />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
