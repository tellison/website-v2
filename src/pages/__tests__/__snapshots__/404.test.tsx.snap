// Vitest Snapshot v1

exports[`404 page > renders correctly 1`] = `
<main>
  <div
    class="container text-center my-5"
  >
    <h1>
      404: Not Found
    </h1>
    <p>
      You just hit a route that doesn't exist... the sadness.
    </p>
  </div>
</main>
`;
